tuxemon/event/conditions/check_evolution.py

```diff
@@ -41,60 +41,75 @@ def test(self, session: Session, condition: MapCondition) -> bool:
-                        _evolution = True
-                    if character.has_tech(evolution.tech):
-                        _evolution = True
-                    if evolution.inside == client.map_inside:
-                        _evolution = True
-                    if evolution.traded == monster.traded:
-                        _evolution = True
-                    if evolution.stats:
-                        params = evolution.stats.split(":")
-                        operator = params[1]
-                        stat1 = monster.return_stat(StatType(params[0]))
-                        stat2 = monster.return_stat(StatType(params[2]))
-                        _evolution = compare(operator, stat1, stat2)
-                    if evolution.variable:
-                        parts = evolution.variable.split(":")
-                        key, value = parts[:2]
-                        if (
-                            key in character.game_variables
-                            and character.game_variables[key] == value
-                        ):
-                            _evolution = True
-                    if evolution.steps:
-                        result = evolution.steps - int(monster.steps)
-                        if result == 0:
-                            _evolution = True
-                            monster.steps += 1
-                        monster.levelling_up = True
-                        monster.got_experience = True
-                    if evolution.bond:
-                        parts = evolution.bond.split(":")
-                        operator, value = parts[:2]
-                        _bond = monster.bond
-                        _evolution = compare(operator, _bond, int(value))
-                    # use the item on the monster
-                    if evolution.item:
-                        _evolution = False
-                    if _evolution:
-                        evolving.append((monster, evolved))
-
-        if evolving:
-            character.pending_evolutions = evolving
-        return len(evolving) > 0
+                    if self.can_evolve(monster, evolution):
+                        evolved_monster = Monster()
+                        evolved_monster.load_from_db(evolution.monster_slug)
+                        evolving_monsters.append((monster, evolved_monster))
+
+        if evolving_monsters:
+            character.pending_evolutions = evolving_monsters
+
+        return len(evolving_monsters) > 0
+
+    def can_evolve(
+        self, monster: Monster, evo: MonsterEvolutionItemModel
+    ) -> bool:
+        # Check if the evolution is actually possible
+        if evo.monster_slug == monster.slug:
+            return False
+
+        conditions = [
+            evo.at_level
+            and compare("greater_or_equal", monster.level, evo.at_level)
+            or False,
+            evo.gender and evo.gender == monster.gender or False,
```

Comment:
Because if any of the attributes (e.g. evo.gender, evo.element, etc.) are missing (i.e., None), the entire condition will evaluate to None, not False.

Do you prefer this?
```
conditions = [
    bool(evo.at_level and compare("greater_or_equal", monster.level, evo.at_level)),
    bool(evo.gender and evo.gender == monster.gender),
    # ...
]
```

Reply:
Yeah, I think that's better.

Ensuring it's a bool isn't strictly necessary, as None is Falsey, but feel free.

Reply:
ok